### ISSUE TITLE
Nexus 5 won't connect to known wifi network after reboot, power off.

### ISSUE THREAD

**Issue by Stefano0101** [ubports/ubports-touch#252]
- Device: Hammerhead
- Channel: 15.04 / rc
- Build: 5

## Steps to reproduce

1. ...Reboot or power off the phone and it won't connect to known wifi network
2. ...Need to select the  wifi network and type the password again
3. ...After another reboot or power off, this needs to be repeated again.

## Expected behavoir

...Phone should remember the wifi network and connect automatically 

## Actual behavior

... but it doesn't

## Logfiles and additional information

... I had noticed that under ,,Previous networks''  there are stored previous networks, but it still won't connect to them, instead, it creates a new entry and gives a number to the network. E.g. my network is called SLX and I found entries SLX, SLX 1, SLX 2, SLX 3, SLX 4, SLX5 ...
So it just can't connect to previously known and connected network and creates a new entry.
This behaviour it's been there for a while now, it's not a new bug introduced by rc/4/5 
Otherwise the wifi connection has no other additional issues.

Let me know if needs any logs.

**Comment by NeoTheThird** [comment]
Thanks for reporting, i can confirm this bug on my device.

**Comment by MK1973** [comment]
Hi, I do not know if I should create new issue or I should add to this one. 

Looks there is a problem with the search engine of available wifi connections. I could not see what my colleagues in their mainstream devices.

**Comment by doniks** [comment]
Happens on Nexus 7 "deb", channel=devel

**Comment by Flohack74** [comment]
Also OPO is affected...

**Comment by stefanomelchior** [comment]
@NeoTheThird Today I tested if bug is still present on hammerhead (15.04/devel), but I can't reproduce it: it seems to work.
I tried at home (wpa2 personal) and at work (wpa enterprise).

**Comment by Stefano0101** [comment]
I tested it today on latest devel and still present. Hammerhead/15.04/devel

**Comment by NeoTheThird** [comment]
I've seen this behavior sometimes on my Opo, but not very often. Most of the time connecting to known WPA2 and WPA2 Enterprise networks works, (same for my N5, actually, which makes reproducing really difficult).

**Comment by Flohack74** [comment]
For me it works... Can this be a special setup of your WiFi? I have an ordinary WPA2 PSK with TKIP etc...

**Comment by NeoTheThird** [comment]
@Stefano0101 What setup do you have?

@Flohack74 Well, even in the same network it sometimes fails, but works most of the time.

I could imagine the issue being caused by too much interference from other networks on the same frequency. I'll use the wifiscanner from the openstore to investigate.

**Comment by bobblkabb** [comment]
I'm on rc and it's still the same.

**Comment by Stefano0101** [comment]
I'm having issue to connect to wpa2. Only after reboot or power off. I might try to do a clean flash and if it helps I'll let you know.

**Comment by Flohack74** [comment]
@NeoTheThird no it seems to be an inherent "misunderstanding" between Network Manager and the lower level parts that sometimes occurs. I had this also already on Ubuntu desktop, but very rarely. Maybe also some device drivers are not happy with the WPA/WPA2 handling of wpa-supplicant and Network Manager. But its really really hard to get any useful logs etc since it just wont work reliable but does not log anything.

**Comment by padraic7a** [comment]
This seems to still be an issue.At least it is happening on my Nexus 5. Wifi network is WPA/WPA2 PSK. 
Is there anything useful I can do?

**Comment by stefanomelchior** [comment]
@padraic7a I usually cancel network setup and I try to connect to again
i.e. This week, after updating to a new release of xenial/devel, it seems not to connect to work wifi on my N5: I cancel its configuration, and I tried to connect again and it worked

**Comment by padraic7a** [comment]
thanks, I'll do that later on and see if I can capture some useful log files.

**Comment by padraic7a** [comment]
this seems to be solved on the Hammerhead 16.04 Devel 20180704/2 build

**Comment by talmai** [comment]
As of OTA-12 I cannot reproduce this.

**Comment by bobblkabb** [comment]
I'm still having the problem on OTA-12.

**Comment by abkro** [comment]
I don't have this problem on my N5 which as of writing runs RC 2020-W19, which should be identical to OTA-12 if I am not mistaken.

**Comment by phisp** [comment]
I don't have this problem with the OTA-15 16.04. Wifi is WPA/WPA2 PSK and it works with TKIP as well.
